build_tools/cmake/iree_cc_library.cmake

```diff
@@ -34,6 +34,7 @@ include(CMakeParseArguments)
 # Also in IDE, target will appear in IREE folder while non PUBLIC will be in IREE/internal.
 # TESTONLY: When added, this target will only be built if user passes -DIREE_BUILD_TESTS=ON to CMake.
 # SHARED: If set, will compile to a shared object.
+# FORCELOAD: If set, links all symbols from "ALWAYSLINK" libraries.
```

Comment:
can we call this something that is easier to map to what it's doing? like `WHOLEARCHIVE`? I wouldn't know what `FORCELOAD` meant without reading this comment line at which time I'd be like 'oh why didn't we just call it that?' :P

Reply:
They are synonyms so I'm fine either way. :)